dynamiqs/hermitian_conjugate.py

```diff
@@ -0,0 +1,23 @@
+from jaxtyping import ArrayLike
+from qutip import Qobj
+
+from .qarrays import QArray, QArrayLike, asqarray
+
+
+class HermitianConjugate:
+    def __radd__(self, y: QArrayLike) -> QArrayLike:
+        if isinstance(y, Qobj):
+            y = asqarray(y)
+
+        if isinstance(y, QArray):
+            return y + y.dag()
+        elif isinstance(y, ArrayLike):
+            return y + y.mT.conj()
+        else:
+            raise TypeError(
+                f'`dq.hq` can only be right-added '
+                f'to a QArrayLike, {y.__class__} given.'
+            )
+
+
+hc = HermitianConjugate()
```

Comment:
We need to document this object in the Python API. Not sure whether we should document the class or this instance (might not be possible to document an instance actually?).

Reply:
I didn't find how to do it, any idea is welcome :)

Reply:
Let me try smthg.

Reply:
It is possible by just putting a docstring under @gautierronan.